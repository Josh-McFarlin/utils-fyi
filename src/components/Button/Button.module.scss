@import "../../../public/static/styles/Colors";

.root {
  max-width: 90%;
  box-sizing: border-box;
  padding: 8px 16px;
  outline: none;
  resize: none;
  white-space: pre-wrap;
  background: $primary-button-color;
  border-width: 1px;
  border-style: solid;
  border-color: $primary-button-color;
  color: $highlight-text-color;
  border-radius: 4px;
  cursor: pointer;
  font-size: 12px;

  &:disabled {
    cursor: not-allowed;
    filter: brightness(80%);
  }

  &:hover {
    filter: brightness(120%);
  }
}

.outlined {
  background: none !important;
}

.primaryColor {
  background: $primary-button-color;
  border-color: $primary-button-color;
}

.secondaryColor {
  background: $secondary-button-color;
  border-color: $secondary-button-color;
}

.successColor {
  background: $success-button-color;
  border-color: $success-button-color;
}

.errorColor {
  background: $error-button-color;
  border-color: $error-button-color;
}

.warningColor {
  background: $warning-button-color;
  border-color: $warning-button-color;
}
