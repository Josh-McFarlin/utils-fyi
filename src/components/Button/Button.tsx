import * as React from "react";
import clsx from "clsx";
import classes from "./Button.module.scss";

interface Props extends React.ButtonHTMLAttributes<HTMLButtonElement> {
  className?: string;
  variant?: "primary" | "secondary" | "success" | "error" | "warning";
  shape?: "filled" | "outlined";
  disabled?: boolean;
}

/**
 * A general button with multiple color variants.
 */
const Button = ({
  className,
  type = "button",
  variant = "primary",
  shape = "filled",
  disabled = false,
  ...rest
}: Props) => (
  <button
    className={clsx(
      classes.root,
      classes[`${variant}Color`],
      shape === "outlined" && classes.outlined,
      className
    )}
    type={type}
    disabled={disabled}
    {...rest}
  />
);

export default Button;
