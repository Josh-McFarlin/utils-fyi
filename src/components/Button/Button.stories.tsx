import * as React from "react";
import Button, { PropTypes } from "./Button";

export default {
  title: "Button",
  component: Button,
  argTypes: {
    variant: {
      type: "select",
      options: ["primary", "secondary", "success", "error", "warning"],
    },
    shape: {
      type: "select",
      options: ["filled", "outlined"],
    },
  },
};

export const Component = (args: PropTypes) => <Button {...args} />;
Component.args = {
  children: "Button",
};

export const Variants = () => (
  <>
    <Button variant="primary">primary</Button>
    <Button variant="secondary">secondary</Button>
    <Button variant="success">success</Button>
    <Button variant="error">error</Button>
    <Button variant="warning">warning</Button>
  </>
);

export const Disabled = (args: PropTypes) => <Button {...args} />;
Disabled.args = {
  children: "Button",
  disabled: true,
};

export const Outlined = (args: PropTypes) => <Button {...args} />;
Outlined.args = {
  children: "Button",
  shape: "outlined",
};
