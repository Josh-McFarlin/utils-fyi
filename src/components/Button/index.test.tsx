import React from "react";
import { render } from "../../../test/testUtils";
import Button from "./Button";

describe("Button Component", () => {
  it("matches snapshot", () => {
    const { asFragment } = render(<Button>Button</Button>, {});

    expect(asFragment()).toMatchSnapshot();
  });
});
