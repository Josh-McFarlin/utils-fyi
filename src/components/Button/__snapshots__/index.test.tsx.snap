// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="root primaryColor"
    type="button"
  >
    Button
  </button>
</DocumentFragment>
`;
